src/main/java/uk/gov/pay/api/model/CreateCardPaymentRequest.java

```diff
@@ -0,0 +1,125 @@
+package uk.gov.pay.api.model;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import io.swagger.annotations.ApiModel;
+import io.swagger.annotations.ApiModelProperty;
+import uk.gov.pay.api.utils.JsonStringBuilder;
+import uk.gov.pay.api.validation.ValidReturnUrl;
+import uk.gov.pay.commons.model.charge.ExternalMetadata;
+
+import javax.validation.Valid;
+import javax.validation.constraints.Size;
+import java.util.Optional;
+import java.util.StringJoiner;
+
+@ApiModel(value = "CreatePaymentRequest", description = "The Payment Request Payload")
+public class CreateCardPaymentRequest extends CreatePaymentRequest {
+
+    public static final String RETURN_URL_FIELD_NAME = "return_url";
+    public static final int URL_MAX_LENGTH = 2000;
+    public static final String PREFILLED_CARDHOLDER_DETAILS_FIELD_NAME = "prefilled_cardholder_details";
+    public static final String PREFILLED_CARDHOLDER_NAME_FIELD_NAME = "cardholder_name";
+    public static final String PREFILLED_BILLING_ADDRESS_FIELD_NAME = "billing_address";
+    public static final String PREFILLED_ADDRESS_LINE1_FIELD_NAME = "line1";
+    public static final String PREFILLED_ADDRESS_LINE2_FIELD_NAME = "line2";
+    public static final String PREFILLED_ADDRESS_CITY_FIELD_NAME = "city";
+    public static final String PREFILLED_ADDRESS_POSTCODE_FIELD_NAME = "postcode";
+    public static final String PREFILLED_ADDRESS_COUNTRY_FIELD_NAME = "country";
+    public static final String DELAYED_CAPTURE_FIELD_NAME = "delayed_capture";
+    public static final String METADATA = "metadata";
+    private static final String PREFILLED_CARDHOLDER_DETAILS = "prefilled_cardholder_details";
+    private static final String BILLING_ADDRESS = "billing_address";
+
+    @ValidReturnUrl
+    @Size(max = URL_MAX_LENGTH, message = "Must be less than or equal to {max} characters length")
+    @JsonProperty("return_url")
+    private final String returnUrl;
+
+    private final Boolean delayedCapture;
+
+    private final ExternalMetadata metadata;
+
+    @Valid
+    private final PrefilledCardholderDetails prefilledCardholderDetails;
+
+    public CreateCardPaymentRequest(CreatePaymentRequestBuilder createPaymentRequestBuilder) {
+        super(createPaymentRequestBuilder);
+        this.prefilledCardholderDetails = createPaymentRequestBuilder.getPrefilledCardholderDetails();
+        this.delayedCapture = createPaymentRequestBuilder.getDelayedCapture();
+        this.returnUrl = createPaymentRequestBuilder.getReturnUrl();
+        this.metadata = createPaymentRequestBuilder.getMetadata();
+    }
+
+    @Override
+    public TokenPaymentType getRequestType() {
+        return TokenPaymentType.CARD;
+    }
+
+    @ApiModelProperty(value = "service return url", required = true, example = "https://service-name.gov.uk/transactions/12345")
+    public Optional<String> getReturnUrl() {
+        return Optional.ofNullable(returnUrl);
+    }
+
+    public Optional<String> getAgreementId() {
```

Comment:
Does this need to be here at all?

Reply:
No, left over from when it was abstract on the base class.